core/services/relay/evm/dual_contract_transmitter.go

```diff
@@ -0,0 +1,181 @@
+package evm
+
+import (
+	"context"
+	"database/sql"
+	"encoding/hex"
+	errors2 "errors"
+	"fmt"
+	"strings"
+
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/common"
+	gethcommon "github.com/ethereum/go-ethereum/common"
+	"github.com/pkg/errors"
+
+	"github.com/smartcontractkit/libocr/offchainreporting2plus/chains/evmutil"
+	ocrtypes "github.com/smartcontractkit/libocr/offchainreporting2plus/types"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/logger"
+
+	"github.com/smartcontractkit/chainlink/v2/core/chains/evm/logpoller"
+)
+
+// TODO: Remove when new dual transmitter contracts are merged
+var dtABI = `[{"inputs":[{"internalType":"bytes32[3]","name":"reportContext","type":"bytes32[3]"},{"internalType":"bytes","name":"report","type":"bytes"},{"internalType":"bytes32[]","name":"rs","type":"bytes32[]"},{"internalType":"bytes32[]","name":"ss","type":"bytes32[]"},{"internalType":"bytes32","name":"rawVs","type":"bytes32"}],"name":"transmitSecondary","outputs":[],"stateMutability":"nonpayable","type":"function"}]`
+
+var _ ContractTransmitter = &dualContractTransmitter{}
+
+type dualContractTransmitter struct {
+	contractAddress     gethcommon.Address
+	contractABI         abi.ABI
+	dualTransmissionABI abi.ABI
+	transmitter         Transmitter
+	transmittedEventSig common.Hash
+	contractReader      contractReader
+	lp                  logpoller.LogPoller
+	lggr                logger.Logger
+	// Options
+	transmitterOptions *transmitterOps
+}
+
+func dualTransmissionABI() abi.ABI {
```

Comment:
I only see one usage, so the current state is fine, but you could simplify a bit (drop the field) and tighten this up by using a https://pkg.go.dev/sync#OnceValue
```suggestion
var dualTransmissionABI = sync.OnceValue(func() abi.ABI {
```
It will even repeat panics, which preserves the current behavior:
> If f panics, the returned function will panic with the same value on every call.

Reply:
Updated in [`71170dc` (#15358)](https://github.com/smartcontractkit/chainlink/pull/15358/commits/71170dc9768506955b5a1a310ec5b1ee46fdaa1f)